Userland/Applications/PixelPaint/Tools/BrushTool.h

```diff
@@ -21,9 +23,17 @@ class BrushTool : public Tool {
     virtual void on_mousemove(Layer*, MouseEvent&) override;
     virtual void on_mouseup(Layer*, MouseEvent&) override;
     virtual GUI::Widget* get_properties_widget() override;
-    virtual Variant<Gfx::StandardCursor, NonnullRefPtr<Gfx::Bitmap>> cursor() override { return Gfx::StandardCursor::Crosshair; }
+    virtual Variant<Gfx::StandardCursor, NonnullRefPtr<Gfx::Bitmap>> cursor() override { return m_cursor; }
 
-    void set_size(int size) { m_size = size; }
+    void set_size(int size)
+    {
+        if (size == m_size)
+            return;
+        m_size = size;
+        m_cursor = build_cursor();
+        if (m_editor)
+            m_editor->update_tool_cursor();
+    }
```

Comment:
I think this snippet might be just big enough that it should go in the `.cpp` file.